app/labor/markdown_parser.rb

```diff
@@ -178,16 +178,24 @@ def escape_liquid_tags_in_codeblock(content)
 
   def wrap_mentions_with_links!(html)
     html_doc = Nokogiri::HTML(html)
-    html_doc.xpath("//body/*[not (@class='highlight')]").each do |el|
-      el.children.each do |child|
-        if child.text?
-          new_child = child.text.gsub(/\B@[a-z0-9_-]+/i) do |s|
+    # looks for node that isn't <code>, <a>, and contains "@"
+    html_doc.xpath('//*[not (self::code) and not(self::a) and contains(text(), "@")]').each do |node|
+      # if the target node is a <p>, <ul>, or <li>, it will have more than 1 child
+      # otherwise inner_html can be use when there's only 1 child
+      if node.children.count > 1
+        # only focus on portion of text with "@"
+        node.xpath("text()[contains(.,'@')]").each do |el|
+          el.replace(el.text.gsub(/\B@[a-z0-9_-]+/i) do |s|
             user_link_if_exists(s)
-          end
-          child.replace(new_child) if new_child != child.text
+          end)
+        end
+      else
+        node.inner_html = node.inner_html.gsub(/\B@[a-z0-9_-]+/i) do |s|
```

Comment:
could you add an inline explanatory comment about this part as well?

Reply:
Will do!